### PR TITLE
ALOOKUP module

### PR DESCRIPTION
**Pull request by bvandersloot**
Adding ALOOKUP module using flags ipv4-lookup and ipv6-lookup

### CODE REVIEW

**Comment by zakird** [comment]
Can you paste an example output?

File: modules/alookup/alookup.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+// Global Factory =============================================================
+//
+type GlobalLookupFactory struct {
+	zdns.BaseGlobalLookupFactory
+	IPv4Lookup bool
+	IPv6Lookup bool
+}
+
+func (s *GlobalLookupFactory) AddFlags(f *flag.FlagSet) {
+	f.BoolVar(&s.IPv4Lookup, "ipv4-lookup", false, "perform A lookups for each server")
```
Comment (zakird) [inline modules/alookup/alookup.go]: Do we want to make this True by default? It's a bit odd to have the default behavior be no lookups.

**Comment by bvandersloot** [comment]
{"name":"www.google.com","status":"success","data":{"exchanges":[{"name":"www.google.com","ipv6_addresses":["2607:f8b0:400e:c03::69"]},{"name":"www.google.com","ipv4_addresses":["74.125.25.106","74.125.25.99","74.125.25.105","74.125.25.103","74.125.25.147","74.125.25.104"]}]}}

**Comment by zakird** [comment]
I don't think this is quite what we want. Why is there exchanges instead of just something like:

`{"name":"www.google.com","status":"success","data":{"ipv6_addresses":["2607:f8b0:400e:c03::69"], "ipv4_addresses":["74.125.25.106","74.125.25.99","74.125.25.105","74.125.25.103","74.125.25.147","74.125.25.104"]}
`

**Comment by bvandersloot** [comment]
I was mirroring the structure of mxlookup. Exchanges will contain a list of the domain names that have an answer in the requests, e.g. if there is a CNAME.

**Comment by bvandersloot** [comment]
Do we want to default to IPv4 or IPv4 and IPv6 ?

**Comment by zakird** [comment]
Exchanges are specified to MX. They don't really make sense in this context. I believe that a CNAME can only point to one domain, so we don't really need a list.

**Comment by zakird** [comment]
I'm not 100% sure. I'd personally go IPv4, but am not set on it.

**Comment by zakird** [comment]
Can you paste the new output again?

**Comment by bvandersloot** [comment]
`{"name":"www.google.com","status":"success","data":{"ipv4_addresses":["74.125.25.105","74.125.25.147","74.125.25.104","74.125.25.99","74.125.25.103","74.125.25.106"],"ipv6_addresses":["2607:f8b0:400e:c03::68"]}}`

**Comment by zakird** [comment]
👍

**Pull request merged**